ansible/ssl_automation.yml

```diff
@@ -14,58 +14,102 @@
-      register: epel_repofile_result
+      # register: epel_repofile_result
 
     - name: Cerbot | Install EPEL repo.
       yum:
         name: "{{ epel_repo_url }}"
         state: present
-      register: result
-      when: not epel_repofile_result.stat.exists
+      # when: not epel_repofile_result.stat.exists
 
     - name: Cerbot | Import EPEL GPG key.
       rpm_key:
         key: "{{ epel_repo_gpg_key_url }}"
         state: present
-      when: not epel_repofile_result.stat.exists
+      # when: not epel_repofile_result.stat.exists
+      # register: epel_repo
 
     - name: Cerbot | Install Cerbot
       yum:
         name: certbot
         state: present
-      register: certbot_installed
+      # when: epel_repo.changed
+      # register: certbot_installed
 
     - name: Certbot | Stop VPN service
       systemd:
         name: openvpnas
         state: stopped
 
+    - name: Certbot | Check if the certs exists in S3
+      shell: aws s3 ls s3://{{ s3_bucket }}/certs/{{ openvpn_server_domain }}/cert.pem | wc -l
+      register: fileCount
+
+    - set_fact:
+        s3fileCount: "{{ fileCount.stdout_lines|length }}"
+
+    - name: Create certificate path
+      file:
+        path: /etc/letsencrypt/live/{{ openvpn_server_domain }}/
+        state: directory
+        mode: '0755'
+        recurse: yes
+      when: fileCount|length > 3
+
+    - name: Certbot | Download S3 certificate files
+      command: aws s3 cp s3://{{ s3_bucket }}/certs/{{ openvpn_server_domain }}/cert.pem /etc/letsencrypt/live/{{ openvpn_server_domain }}/cert.pem
+      when: fileCount|length > 3
+
+    - name: Certbot | Download S3 certificate files
+      command: aws s3 cp s3://{{ s3_bucket }}/certs/{{ openvpn_server_domain }}/chain.pem /etc/letsencrypt/live/{{ openvpn_server_domain }}/chain.pem
+      when: fileCount|length > 3
+
+    - name: Certbot | Download S3 certificate files
+      command: aws s3 cp s3://{{ s3_bucket }}/certs/{{ openvpn_server_domain }}/fullchain.pem /etc/letsencrypt/live/{{ openvpn_server_domain }}/fullchain.pem
+      when: fileCount|length > 3
+
+    - name: Certbot | Download S3 certificate files
+      command: aws s3 cp s3://{{ s3_bucket }}/certs/{{ openvpn_server_domain }}/privkey.pem /etc/letsencrypt/live/{{ openvpn_server_domain }}/privkey.pem
+      when: fileCount|length > 3
+      
     - name: Cerbot | Request certificate
-      command: certbot certonly --standalone -d {{ openvpn_server_domain }} --email {{ ssl_admin_email }} --agree-tos
+      command: certbot certonly --standalone -d {{ openvpn_server_domain }} --email {{ ssl_admin_email }} --agree-tos --non-interactive
       register: certificate_request
+      when: s3fileCount|length == 0
 
     - name: Certbot | Start VPN service
       systemd:
         name: openvpnas
         state: restarted
       when: certificate_request.changed
 
+    - name: Rename certificate directory
+      command: mv vpn-* vpn-test.eks.33n.co.uk
```

Comment:
@richardgutkowski Do we need to rename / move this to a variable?

Reply:
Updated to `mv {{ domain_name }}-* {{ domain_name}}` the reason for this step is to work around the auto-generated `-00X` suffix that is now added to domain names by Certbot when running it in hands-off mode.